client/src/components/forms/SignUp/index.tsx

```diff
@@ -1,47 +1,113 @@
-import React, { useState } from 'react';
-import { signup } from 'utils/data/user';
-import * as M from 'utils/network/errorMessages';
+import React from 'react';
+import {
+  Typography,
+  Theme,
+  createStyles,
+  WithStyles,
+  withStyles,
+  TextField,
+  Box,
+  Divider,
+  Select,
+  MenuItem,
+  FormControl,
+  InputLabel,
+} from '@material-ui/core';
+import BWButton from 'components/buttons/BWButton';
+import { FACULTIES, SEMESTERS } from 'utils/constants';
 
-function SignUp() {
-  const [name, setName] = useState('');
-  const [email, setEmail] = useState('');
-  const [password, setPassword] = useState('');
+const styles = (theme: Theme) =>
+  createStyles({
+    titleDivider: {
+      marginTop: theme.spacing(1),
+      backgroundColor: theme.palette.text.primary,
+    },
+    form: {
+      marginTop: theme.spacing(2),
+      padding: `${theme.spacing(1)}px ${theme.spacing(2)}px`,
+    },
+    actions: {
+      marginTop: theme.spacing(2),
+      padding: `0 ${theme.spacing(4)}px`,
+    },
+  });
 
-  const submit = async () => {
-    try {
-      const { user, token } = await signup({ email, password, name });
-      console.log('Sign up successful', user, token);
-    } catch (error) {
-      console.log(M.SIGN_UP, error);
-    }
-  };
+type Props = WithStyles<typeof styles>;
 
+function SignUp({ classes }: Props) {
   return (
     <div>
-      <h1>Sign Up</h1>
-      <p>Name</p>
-      <input
-        value={name}
-        onChange={(e) => setName(e.target.value)}
-        type="text"
-      />
-      <p>Email</p>
-      <input
-        value={email}
-        onChange={(e) => setEmail(e.target.value)}
-        type="email"
-      />
-      <p>Password</p>
-      <input
-        value={password}
-        onChange={(e) => setPassword(e.target.value)}
-        type="password"
-      />
-      <div style={{ marginTop: 10 }}>
-        <button onClick={submit}>Create</button>
+      <Typography variant="h4" align="center">
+        Enter Your Information
+      </Typography>
+      <Divider className={classes.titleDivider} />
+      <div className={classes.form}>
+        <Box display="flex">
+          <Box flex={1}>
```

Comment:
I like these [`Box`s](https://material-ui.com/components/box/) for small layout styles so I don't have to define a whole class

Reply:
Damn that's pretty sick. I think i'll start using them too